component-interaction/src/app/components/user-list.component/user-list.component.ts

```diff
@@ -10,10 +10,34 @@ import { UserService } from "src/app/services/user.service";
 })
 export class UserListComponent implements OnInit{
     public users?: UserModel[];
+ 
 
     constructor(private userService: UserService){}
     ngOnInit(): void {
         this.users = this.userService.getUsers();
     }
 
+    addUser(): void{
+        let email = prompt("Enter Your Email");
+        if(email == null){ // deoarece prompt poate intoarce null, asignez empty
+            email = "";   // string in acest caz pentru a nu avea erori
+        }                // in cazul in care avem empty string, user-ul nu va fi creat si se va afisa o alerta
+        let name = prompt("Enter Your Name");
+        if(name == null){
+            name = "";
+        }
+        if(email.length > 0 && name.length > 0){
+            let newUser: UserModel = {
+                "email": email,
+                "dateOfBirth": new Date(),
+                "name": name
+            };
+            this.userService.addUser(newUser);
+        }else{
+            alert("The New User Could Not Be Added");
+        }
+        
```

Comment:
```suggestion
        let email = prompt("Enter Your Email");
        if (!email) { // deoarece prompt poate intoarce null, asignez empty
            email = "";   // string in acest caz pentru a nu avea erori
        }                // in cazul in care avem empty string, user-ul nu va fi creat si se va afisa o alerta
        let name = prompt("Enter Your Name");
        if (!name) {
            name = "";
        }
        if (email.length === 0 || name.length === 0) {
          alert("The New User Could Not Be Added");
          return;
        }
        const newUser: UserModel = {
                "email": email,
                "dateOfBirth": new Date(),
                "name": name
         };
         this.userService.addUser(newUser);
```

Reply:
you can do something like this. Instead you have an if with some instructions that will be done, try to have an negative if that will exit the function